device_doctor/bin/main.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:args/args.dart';
+
+import 'package:device_doctor/src/config.dart';
+
+const String actionFlag = 'action';
+const String deviceOSFlag = 'deviceOS';
+const String helpFlag = 'help';
+
+/// These values will be initialized in `_checkArgs` function,
+/// and used in `main` function.
+String _action;
+String _deviceOS;
+
+/// Manage healthcheck and recovery for devices.
+///
+/// Usage:
+/// dart main.dart --action <healthcheck|recovery> --deviceOS <android|ios>
+Future<void> main(List<String> args) async {
+  final ArgParser parser = ArgParser();
+  parser
+    ..addOption('$actionFlag')
+    ..addOption('$deviceOSFlag')
+    ..addFlag('$helpFlag');
+
+  if (!_checkArgs(parser, args)) {
+    stdout.write('\nRequired flags:\n'
+        '--$actionFlag Supported actions are healthcheck and recovery.\n'
```

Comment:
Arg parser can be used to generate this dynamically. The devicelab test runner does it that way - https://github.com/flutter/flutter/blob/master/dev/devicelab/bin/run.dart

Reply:
Thanks for this info. Updated the parser with `help`.